ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import errorMessage from 'vault/utils/error-message';
+
+import type ConfigModel from 'vault/models/azure/config';
+import type IdentityOidcConfigModel from 'vault/models/identity/oidc/config';
+import type Router from '@ember/routing/router';
+import type StoreService from 'vault/services/store';
+import type VersionService from 'vault/services/version';
+import type FlashMessageService from 'vault/services/flash-messages';
+
+/**
+ * @module ConfigureAzureComponent is used to configure the Azure secret engine
+ * For enterprise users, they will see an additional option to config WIF attributes in place of Azure account attributes.
+ * If the user is configuring WIF attributes they will also have the option to update the global issuer config, which is a separate —global— endpoint named identity/oidc/config.
+ * @example
+ * ```js
+ * <SecretEngine::ConfigureAzure
+    @model={{this.model.azure-config}}
+    @backendPath={{this.model.id}}
+    @issuerConfig={{this.model.identity-oidc-config}}
+    />
+ * ```
+ *
+ * @param {object} model - Azure config model
+ * @param {string} backendPath - name of the Azure secret engine, ex: 'azure-123'
+ * @param {object} issuerConfigModel - the identity/oidc/config model
+ */
+
+interface Args {
+  model: ConfigModel;
+  issuerConfig: IdentityOidcConfigModel;
+  backendPath: string;
+}
+
+export default class ConfigureAwsComponent extends Component<Args> {
+  @service declare readonly router: Router;
+  @service declare readonly store: StoreService;
+  @service declare readonly version: VersionService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked errorMessage: string | null = null;
+  @tracked invalidFormAlert: string | null = null;
+  @tracked accessType = 'azure';
+  @tracked saveIssuerWarning = '';
+
+  disableAccessType = false;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+
+    if (this.version.isCommunity || this.args.model.isNew) return; // the following checks are relevant only to enterprise users and those editing an existing configuration.
```

Comment:
So this means we'll return when a user on community is editing, is that correct? I guess this would be slightly clearer to me in the reverse, which would mean we could also remove the comment: 
```suggestion
    if (this.version.isEnterprise && !this.args.model.isNew) {
    ....logic below
    
```